### PR TITLE
- Xcode 9 Swift 4.0 Support

### PR DESCRIPTION
**Pull request by mosn**
Adds support for swift 4.0

### CODE REVIEW

**Comment by mosn** [comment]
Maybe master is not the right branch, so please change it to your preferred branch

File: InteractiveSideMenu.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -266,13 +266,13 @@
 				DYLIB_INSTALL_NAME_BASE = "@rpath";
 				INFOPLIST_FILE = "$(SRCROOT)/Sources/Info.plist";
 				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
-				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
```
Comment (AndreyArzhannikov) [inline InteractiveSideMenu.xcodeproj/project.pbxproj]: @mosn Could you clarify why we need to change `IPHONEOS_DEPLOYMENT_TARGET` to 10.0?

File: Sources/MenuAnimator.swift [hunk 2]
```diff
@@ -20,7 +20,7 @@ import UIKit
 
 /**
  Options that define menu transition settings.
- Class lets user customize their transitioning animation: duration, content scale, visible content width, animationOptions, using spring animation or not and params if yes.
+ Class lets user customize their transitioning animation: duration, content scale, visible content width, animationOptions, using spring animation or not and params if ye@objc s.
```
Comment (AndreyArzhannikov) [inline Sources/MenuAnimator.swift]: @mosn There is a typo `if ye@objc s.`. Could you fix it?

File: Sources/MenuViewController.swift [hunk 3]
```diff
@@ -22,7 +22,8 @@ open class MenuViewController: UIViewController {
     
     public weak var menuContainerViewController: MenuContainerViewController?
     var navigationMenuTransitionDelegate: MenuTransitioningDelegate?
-        
+
+    @objc
```
Comment (AndreyArzhannikov) [inline Sources/MenuViewController.swift]: @mosn please put this annotation on the same line as the function declaration.

File: Sources/MenuAnimator.swift [hunk 4]
```diff
@@ -219,14 +219,15 @@ class MenuInteractiveTransition: NSObject, UIViewControllerInteractiveTransition
         
         finishTransition(currentPercentComplete: 0)
     }
-    
+
+    @objc
```
Comment (AndreyArzhannikov) [inline Sources/MenuAnimator.swift]: @mosn please put this annotation on the same line as the function declaration.

**Comment by AndreyArzhannikov** [comment]
Hi, @mosn.

Thanks for contributing! We really appreciate it.

About changes, we think that is still early for updating production branch to swift 4 since it's in beta but it makes sense for develop branch taking into account fixing comments.

So, follow this steps for contributing, please:
- To be more confident with contributing please get familiar with [CONTRIBUTING](https://github.com/handsomecode/InteractiveSideMenu/blob/master/CONTRIBUTING.md) and 
[CODE_OF_CONDUCT](https://github.com/handsomecode/InteractiveSideMenu/blob/master/CODE_OF_CONDUCT.md) guides.
- Fix current comments
- Change target branch from `master` to `develop`.

**Comment by AndreyArzhannikov** [comment]
It is a duplicate of these #44 and #47.

**Pull request closed without merging**